### PR TITLE
Update dependencies, add x-checker for everything, remove useless stuff

### PR DESCRIPTION
**Pull request by lionirdeadman**
Please test!

### CODE REVIEW

**Comment by oxy** [comment]
works on my machine! :+1:

**Comment by eiglow** [comment]
Seems to work fine for me.

File: com.discordapp.Discord.json [hunk 1]
```diff
@@ -168,7 +157,12 @@
                         {
                             "type": "archive",
                             "url": "https://github.com/pciutils/pciutils/archive/v3.7.0.tar.gz",
-                            "sha256": "ea768aa0187ba349391c6c157445ecc2b42e7d671fc1ce8c53ff5ef513f1e2ab"
+                            "sha256": "ea768aa0187ba349391c6c157445ecc2b42e7d671fc1ce8c53ff5ef513f1e2ab",
+		            "x-checker-data": {
+		                "type": "anitya",
+		                "project-id": "2605",
+		                "url-template": "https://github.com/pciutils/pciutils/archive/v$version.tar.gz"
+		            }
```
Comment (Lctrs) [inline com.discordapp.Discord.json:165]: indentation

File: com.discordapp.Discord.json [hunk 2]
```diff
@@ -54,22 +54,51 @@
...
                 {
                     "type": "archive",
-                    "url": "https://xorg.freedesktop.org/archive/individual/app/xprop-1.2.4.tar.bz2",
-                    "sha256": "8c77fb096e46c60032b7e2bde9492c3ffcc18734f50b395085a5f10bfd3cf753"
+                    "url": "https://xorg.freedesktop.org/archive/individual/app/xprop-1.2.5.tar.bz2",
+                    "sha256": "9b92ed0316bf2486121d8bac88bd1878f16b43bd335f18009b1f941f1eca93a1",
+                    "x-checker-data": {
+                        "type": "anitya",
+	                "project-id": "14958",
+	                "url-template": "https://xorg.freedesktop.org/archive/individual/app/xprop-$version.tar.bz2"
```
Comment (Lctrs) [inline com.discordapp.Discord.json:62]: indentation

File: com.discordapp.Discord.json [hunk 3]
```diff
@@ -54,22 +54,51 @@
...
+                    }
                 }
             ]
         },
+	{
+	  "name": "python-setuptools_scm",
+	  "buildsystem": "simple",
+	  "build-commands": [
+	    "python3 setup.py install --prefix=/app --root=/"
```
Comment (Lctrs) [inline com.discordapp.Discord.json:71]: Use pip and same args as in https://github.com/flatpak/flatpak-builder-tools/blob/master/pip/flatpak-pip-generator

```suggestion
	    "pip3 install --no-deps --no-build-isolation --verbose --prefix=${FLATPAK_DEST} ."
```

File: com.discordapp.Discord.json [hunk 4]
```diff
@@ -54,22 +54,51 @@
...
+	    }
+	  ]
+	},
         {
-            "name": "python3-XLib",
+            "name": "python-xlib",
             "buildsystem": "simple",
             "build-commands": [
-                "pip3 install --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} XLib"
+                "pip3 install --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} python-xlib"
```
Comment (Lctrs) [inline com.discordapp.Discord.json:91]:
```suggestion
                "pip3 install --no-deps --no-build-isolation --verbose --prefix=${FLATPAK_DEST} ."
```

**Comment by lionirdeadman** [comment]
@Lctrs Thanks for the review! All fixed up.

**Pull request merged**